### DISCUSSION TITLE
rready && rrvalid 提早 handshake 的問題

### DISCUSSION THREAD

**Discussion by chengyu-123** [bol-edu/HLS-SOC-Discussions#92] [Q&A]
我想請教一下，這邊 rready, rvalid 提早 handshake 導致對答案時對到前一筆的問題  ，可以給我一點想法嗎，謝謝!
目前試過把 rvalid 晚3 T 拉起來但也會連帶影響到 rready 拉起來的時間，行不通
<img width="284" alt="image" src="https://github.com/bol-edu/HLS-SOC-Discussions/assets/69706789/2e323679-38b1-4ac9-a0fb-6d359d5b2b20">
<img width="308" alt="image" src="https://github.com/bol-edu/HLS-SOC-Discussions/assets/69706789/6788b6d3-f310-4b9b-86e5-80e88a9e951d">

**Comment by samlo123** [comment]
在得到addr之後的下一個clk會讀出data，同時間也順便把rvalid 拉起來。
應該是這樣理解吧
![Screenshot_20231010-020858.png](https://github.com/bol-edu/HLS-SOC-Discussions/assets/101400844/7c33df34-df1d-4c09-8f0c-951ae69f877e)

**Reply by chengyu-123** [reply]
我了解了~非常感謝你

**Comment by boledulab** [comment]
I have a feeling it may relate to the blocking/nonblock simulation scheduling issue. Check your testbench and RTL code.
